### ISSUE TITLE
[Bug Report] Secondary files being generated unnecessarily after file rename

### ISSUE THREAD

**Issue by Darklyter** [stashapp/stash#4393]
**Describe the bug**
Since file refactor, it seems that scenes that are renamed after having secondary files (preview, scrubber sprites, etc) generated are generating those secondary files again on reimport/rematch by Stash

**To Reproduce**
1) Import a scene
2) Generate preview, scrubber sprites, etc
3) Rename file on hard drive
4) Perform a new scan to include now renamed scene
5) Watch as Stash matches the new filename to the old file entry, but then proceeds to generate a new copy of the preview, sprites, etc even though they already exist

**Expected behavior**
I would expect Stash to realize that generated secondary files (such as previews) already exist, named for the video using OSHASH and skip having to unnecessarily generate new copies.

**Screenshots**
If applicable, add screenshots to help explain your problem please ensure that your screenshots are SFW or at least appropriately censored.

No screenshot, but here is a log snippet.  This file was renamed and moved to a new directory, but in no other way changed from when it was previously imported into Stash and a Generate task run on it:

[36mINFO[0m[2023-12-25 04:11:13] /data/Import Rips/AllJapanesePass/jcosplay - 2022-02-27 - Hot Japanese Teen In Red Panties Kirara Kurokawa Gets Licked And Pounded.mkv moved to /data/Import Rips/AllJapanesePass/All Japanese Pass/Japanese Cosplay/Japanese Cosplay 2005-11-20 Hot Japanese Teen In Red Panties Kirara Kurokawa Gets Licked And Pounded (Junko Hayama) [720x408].mkv. Updating path... 
[36mINFO[0m[2023-12-25 07:11:25] [generator] generating sprite image for /data/Import Rips/AllJapanesePass/All Japanese Pass/Japanese Cosplay/Japanese Cosplay 2005-11-20 Hot Japanese Teen In Red Panties Kirara Kurokawa Gets Licked And Pounded (Junko Hayama) [720x408].mkv 
[36mINFO[0m[2023-12-25 07:11:29] [generator] generating video preview for /data/Import Rips/AllJapanesePass/All Japanese Pass/Japanese Cosplay/Japanese Cosplay 2005-11-20 Hot Japanese Teen In Red Panties Kirara Kurokawa Gets Licked And Pounded (Junko Hayama) [720x408].mkv 
[36mINFO[0m[2023-12-25 07:11:32] [generator] generating webp preview for /data/Import Rips/AllJapanesePass/All Japanese Pass/Japanese Cosplay/Japanese Cosplay 2005-11-20 Hot Japanese Teen In Red Panties Kirara Kurokawa Gets Licked And Pounded (Junko Hayama) [720x408].mkv 
[36mINFO[0m[2023-12-25 07:11:42] [generator] generating sprite vtt for /data/Import Rips/AllJapanesePass/All Japanese Pass/Japanese Cosplay/Japanese Cosplay 2005-11-20 Hot Japanese Teen In Red Panties Kirara Kurokawa Gets Licked And Pounded (Junko Hayama) [720x408].mkv 


**Stash Version: (from Settings -> About):**
Still on v22.1

**Desktop (please complete the following information):**
Running in Docker on Ubuntu

**Comment by WithoutPants** [comment]
I'm not able to reproduce this on v0.22.1 or the latest dev build.

Having looked at the code, the generator only logs those messages either if it has been instructed to overwrite (which it will not do during a scan operation) or if it cannot access the existing generated files. The generator does not discriminate between a missing file or bad permissions etc when determining whether to regenerate.

**Comment by Darklyter** [comment]
Continuation from [Discord ](https://discord.com/channels/559159668438728723/559159668912553989/1208911397384233072)

I matched my Wankz rip (about 2300 scenes) against Stashdb and TPDB via phash and filename.  Phash was created for all of them previously, and was verified by a filter of "Phash is null" being blank.  However on renaming the files and moving them to a different directory, when doing a new scan it went through all of them performing the Calculating fingerprints and Adding functions.  However immediately after that it started recreating the phash sprites for all of the files.

So the re-import has been running for 15 minutes or so, and it's at 66% currently.  Again, the only change made between import 1 and import 2 was the files being renamed and then moved into the "Organized" folder.

[stash_log.txt](https://github.com/stashapp/stash/files/14324826/stash_log.txt)

**Comment by Darklyter** [comment]
By way of explanation, the rename is simply done via a python with a `shutil.move(origfile, filepathname + extension)` command
[file_rename.txt](https://github.com/stashapp/stash/files/14324836/file_rename.txt)

**Comment by WithoutPants** [comment]
An important distinction here is that according to the logs, the file has _not_ been moved. Scan has found a new file, and the old location still exists - hence the log entries `Adding <newpath>.mkv to scene <scene name>`

If the file was moved and scan can no longer find the old path, then it is treated as a move.

There's a couple of improvements to be made here:
- if the old location is not in the configured library paths, then scan needs to treat this as a move
- if a new file is added that has the same oshash as an existing file, then it should probably use its phash instead of regenerating another

I also noticed an `already in transaction` error that occurs during folder rename detection if there are captions in the path.
